src/github.com/couchbase/sync_gateway/rest/config.go

```diff
@@ -256,17 +256,17 @@ func (dbConfig DbConfig) validate() error {
 
 // Implementation of AuthHandler interface for DbConfig
 func (dbConfig *DbConfig) GetCredentials() (string, string, string) {
-	return dbConfig.Username, dbConfig.Password, *dbConfig.Bucket
+	return base.TransformBucketCredentials(dbConfig.Username, dbConfig.Password, *dbConfig.Bucket)
 }
 
 // Implementation of AuthHandler interface for ShadowConfig
 func (shadowConfig *ShadowConfig) GetCredentials() (string, string, string) {
-	return shadowConfig.Username, shadowConfig.Password, *shadowConfig.Bucket
+	return base.TransformBucketCredentials(shadowConfig.Username, shadowConfig.Password, *shadowConfig.Bucket)
 }
 
 // Implementation of AuthHandler interface for ChannelIndexConfig
 func (channelIndexConfig *ChannelIndexConfig) GetCredentials() (string, string, string) {
-	return channelIndexConfig.Username, channelIndexConfig.Password, *channelIndexConfig.Bucket
+	return base.TransformBucketCredentials(channelIndexConfig.Username, channelIndexConfig.Password, *channelIndexConfig.Bucket)
```

Comment:
If we're doing this for dbConfig, shadowConfig, and channelIndexConfig, can we just get rid of dcpAuthHandler altogether?  Right now it seems like when we instantiate an auth handler using getDcpAuthHandler (https://github.com/couchbase/sync_gateway/blob/feature/distributed_index/src/github.com/couchbase/sync_gateway/base/bucket.go#L318), we're going to be doing the transform twice - once here:

https://github.com/couchbase/sync_gateway/blob/feature/distributed_index/src/github.com/couchbase/sync_gateway/base/bucket.go#L323

and then later when we call dcpAuthHandler.GetCredentials() - such as here:

https://github.com/couchbase/sync_gateway/blob/feature/distributed_index/src/github.com/couchbase/sync_gateway/base/bucket.go#L170